state/cleanup.go

```diff
@@ -415,15 +425,25 @@ func (st *State) cleanupMachinesForDyingModel(cleanupArgs []bson.Raw) (err error
 		} else {
 			err = m.Destroy()
 		}
-		if err != nil {
+		if err != nil && IsHasContainersError(err) {
+			// If we do not continue, the containers will never be destroyed,
+			// thus, this machine will not be.
+			continueError = true
+			logger.Warningf("%v", err)
+		} else if err != nil {
 			err = errors.Annotatef(err, "while destroying machine %v is", m.Id())
-			// TODO (force 2019-4-24) we should not break out here but continue with other machines.
 			if !force {
 				return errors.Trace(err)
 			}
+			continueError = true
 			logger.Warningf("%v", err)
 		}
 	}
+	if continueError {
+		// We need to retry the failed destroys.  The errors have been
+		// logged, return an error here so we try again later.
+		return errors.New("Non fatal failure(s) found during machine destroy, try again later")
```

Comment:
This is a user error message at the wrong layer. cleanup jobs are run by a worker and errors are not exposed to the user.

Reply:
This was not intended as a specific message for users, rather the method needs to return an error so that it will be retried.  I could have picked one of n potential errors produced instead.